app/workers/deletes.php

```diff
@@ -245,25 +249,56 @@ protected function deleteMemberships(Document $document, string $projectId): voi
 
     /**
      * @param Document $document project document
+     * @throws Exception
      */
     protected function deleteProject(Document $document): void
     {
         $projectId = $document->getId();
 
-        // Delete all DBs
-        $this->getProjectDB($projectId)->delete($projectId);
+        // Delete project tables
+        $dbForProject = $this->getProjectDB($projectId, $document);
+
+        $limit = 50;
+        $offset = 0;
+
+        while (true) {
+            $collections = $dbForProject->listCollections($limit, $offset);
+
+            if (empty($collections)) {
+                break;
+            }
+
+            foreach ($collections as $collection) {
+                $dbForProject->deleteCollection($collection->getId());
+            }
+
+            $offset += $limit;
+        }
+
+        // Delete metadata tables
+        try {
+            $dbForProject->deleteCollection('_metadata');
+        } catch (Exception) {
+            // Ignore: deleteCollection tries to delete a metadata entry after the collection is deleted,
+            // which will throw an exception here because the metadata collection is already deleted.
+        }
 
         // Delete all storage directories
-        $uploads = $this->getFilesDevice($document->getId());
-        $cache = new Local(APP_STORAGE_CACHE . '/app-' . $document->getId());
+        $uploads = $this->getFilesDevice($projectId);
+        $functions = $this->getFunctionsDevice($projectId);
+        $builds = $this->getBuildsDevice($projectId);
+        $cache = $this->getCacheDevice($projectId);
```

Comment:
Don't we also need to delete the `config` and `certificates` associated with a project? 
https://github.com/appwrite/appwrite/blob/master/app/init.php#L114-L115

Reply:
Both are associated with an installation instead of a project :+1:

Reply:
Yes, but we still need to delete the certificates of the domains associated with a project right ?

Reply:
Ah yep my mistake!